### PR TITLE
Add support for JWT RFC7519 registered claims [fixes #70]

### LINKED ISSUE #70: JWT: Add support for `iss` and `aud` claims
These claims are required by some service proxies that supports JWS verification.
Unfortunately claims cannot be injected using a custom serialiser in the application code, since its output is nested together with other payload fields under the `data` key.

### PR DESCRIPTION
**Pull request by mkubala**
This PR brings support for all registered claims mentioned in [RFC 7519 section 4.1 ](https://tools.ietf.org/html/rfc7519#page-9).
Also it solves #70 

Readme will be updated in a separate commit.

### CODE REVIEW

File: core/src/main/scala/com/softwaremill/session/SessionConfig.scala [hunk 1]
```diff
@@ -70,14 +73,22 @@ object SessionConfig {
       if (config.getAnyRef(path) == noneValue) None
       else
         Some(config.getDuration(path, TimeUnit.SECONDS))
+    def getStringIfExists(path: String): Option[String] = ifExists(path, config.getString)
+    def getDurationSecondsIfExists(path: String): Option[Long] = ifExists(path, config.getDuration(_, TimeUnit.SECONDS))
+    def getBooleanIfExists(path: String): Option[Boolean] = ifExists(path, config.getBoolean)
+    def getConfigIfExists(path: String): Option[Config] = ifExists(path, config.getConfig)
+
+      private def ifExists[T](key: String, getter: String => T): Option[T] =
```
Comment (magdzikk) [inline core/src/main/scala/com/softwaremill/session/SessionConfig.scala]: Too much indentation?

**Review by magdzikk — APPROVED** [review]

File: core/src/main/scala/com/softwaremill/session/SessionConfig.scala [hunk 2]
```diff
@@ -91,6 +102,18 @@ object SessionConfig {
             throw new IllegalArgumentException(s"Unsupported JWS alg '$oth'. Supported algorithms are: HS256, RS256")
         }
       },
+      jwt = {
+        val claimsConfig = scopedConfig.getConfig("jwt")
+
+        JwtConfig(
+          issuer = claimsConfig.getStringIfExists("iss"),
```
Comment (kciesielski) [inline core/src/main/scala/com/softwaremill/session/SessionConfig.scala:109]: Why such short config keys?

File: jwt/src/main/scala/com/softwaremill/session/JwtSessionEncoder.scala [hunk 3]
```diff
@@ -67,23 +67,42 @@ class JwtSessionEncoder[T](implicit serializer: SessionSerializer[T, JValue], fo
...
+      // and https://tools.ietf.org/html/rfc7519#page-6
+      val exp = numericDateClaimFromTimeout("exp", expirationTimeout).toList
+      val nbf = numericDateClaimFromTimeout("nbf", notBeforeTimeout).toList
+      val iat = if (includeIssuedAt) numericDateClaimFromTimeout("iat", Some(0L)).toList else Nil
+
+      // 'jti' must be unique per token,
+      // collisions MUST be prevented even among values produced by different issuers,
+      // see https://tools.ietf.org/html/rfc7519#page-10
+      val jti = if (includeRandomJwtId) {
+        stringClaim("jti", Some(issuer.map(_ + "-").getOrElse("") + UUID.randomUUID().toString)).toList
```
Comment (kciesielski) [inline jwt/src/main/scala/com/softwaremill/session/JwtSessionEncoder.scala:93]: Just so you are aware:  `UUID.randomUUID()` may cause performance issues on heavy load. For such cases one can use for example `com.eatthepath:fast-uuid`. However, I don't think it's necessary here.

File: core/src/main/scala/com/softwaremill/session/SessionConfig.scala (hunk 2 above)
Comment (kciesielski) [reply core/src/main/scala/com/softwaremill/session/SessionConfig.scala:109]: Ok I guess this is because these should match session keys.

Comment (mkubala) [reply core/src/main/scala/com/softwaremill/session/SessionConfig.scala:109]: Exactly, they reflect RFC 7519 registered claim identifiers.

**Pull request merged**